### PR TITLE
Support C# apps & deps

### PR DESCRIPTION
**Pull request by duncanjbrown**
Depends on #44.

Support C# apps in the list of apps and dependencies.

Refactors and adds some basic tests for the existing ruby repo fetching code and makes a C# variant, also tested.

Screenshot 2023-05-26 at 14 27 26

### CODE REVIEW

File: config/services.yml [hunk 1]
```diff
@@ -22,80 +20,131 @@
...
+  docsets:
+    - path: "adr"
+    - path: "docs"
+- name: "Refer serious misconduct by a teacher"
+  repo_name: "DFE-Digital/refer-serious-misconduct"
+  language: 'ruby'
+  docsets:
+    - path: "adr"
+    - path: "docs"
+- name: "Qualified teachers API"
```
Comment (gunndabad) [inline config/services.yml:131]: This repo got renamed last week, it's now at `teaching-record-system`.

File: config/services.yml [hunk 2]
```diff
@@ -22,80 +20,131 @@
...
+  repo_name: "DFE-Digital/qualified-teachers-api"
+  language: 'cs'
+  csproj_path: >-
+    QualifiedTeachersApi/src/QualifiedTeachersApi/QualifiedTeachersApi.csproj
```
Comment (gunndabad) [inline config/services.yml:135]:
```suggestion
    TeachingRecordSystem/src/TeachingRecordSystem.Api/TeachingRecordSystem.Api.csproj
```

Comment (duncanjbrown) [reply config/services.yml:135]: Brilliant 🙏

**Comment by duncanjbrown** [comment]
Amended

**Review by gunndabad — APPROVED** [review]

**Pull request merged**